src/main/java/org/opensearch/knn/indices/ModelCache.java

```diff
@@ -65,9 +68,9 @@ public synchronized void rebuild() {
     }
 
     protected ModelCache() {
-        cacheSizeInBytes = MODEL_CACHE_SIZE_IN_BYTES_SETTING.get(clusterService.getSettings());
-        clusterService.getClusterSettings().addSettingsUpdateConsumer(MODEL_CACHE_SIZE_IN_BYTES_SETTING, it -> {
-            cacheSizeInBytes = it;
+        cacheSizeInKB = MODEL_CACHE_SIZE_LIMIT_SETTING.get(clusterService.getSettings()).getKb();
```

Comment:
Are we sure if its Kilobits or KiloBytes?  We seem to do ```.getkb```?

Reply:
Wrong function. This is the right one:
```
    public long getKb() {
        return this.unit.toKB(this.size);
    }

```

Here they use KB to get Kilobytes